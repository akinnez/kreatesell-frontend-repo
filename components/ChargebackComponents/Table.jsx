import { EmptyDataTable } from "utils";
import styles from "./Table.module.scss";
import Image from "next/image";
import { IoEllipsisHorizontal } from "react-icons/io5";
import { Popover } from "antd";
import { useState } from "react";
import { AiOutlineClose, AiOutlineEye } from "react-icons/ai";
import { BsReplyFill } from "react-icons/bs";
import moment from "moment";

const SuccessDiv = ({ text }) => {
  return (
    <div
      style={{
        background: "#4bf71625",
        borderRadius: "5px",
        width: "50px",
        textAlign: "center",
      }}
    >
      <p style={{ color: "#176300" }}>{text}</p>
    </div>
  );
};

const ErrorDiv = ({ text }) => {
  return (
    <div
      style={{
        background: "#f7211622",
        borderRadius: "5px",
        width: "50px",
        textAlign: "center",
      }}
    >
      <p style={{ color: "#f5222d" }}>{text}</p>
    </div>
  );
};

export const Table = ({ header, data, loading }) => {
  const [openPopOver, setOpenPopOver] = useState(false);
  return (
    <div className={styles.tableContainer}>
      <table className={styles.table}>
        <thead>
          <tr>
            {header?.map((item, i) => (
              <th key={i}>{item?.title}</th>
            ))}
          </tr>
        </thead>

        {Boolean(data?.length) && (
          <tbody className="t-body">
            {data?.map((item, i) => (
              <tr key={item?.id || i}>
                <td>{item.id}</td>
                <td>{item.name}</td>
                <td>{item.desc}</td>
                <td>{item.department}</td>
                <td>{moment(item.date).format("YYYY-MM-DD HH:mm:ss")}</td>
                <td>
                  {item.status ? (
                    <ErrorDiv text="Open" />
                  ) : (
                    <SuccessDiv text="Closed" />
                  )}
                </td>
                <td>
                  <Popover
                    placement="bottomRight"
                    title={null}
                    content={
                      <u className={styles.popoverUl}>
                        <li className={styles.popoverLi}>
                          <AiOutlineEye className={styles.popoverIcon} />
                          <p>View Ticket</p>
                        </li>
                        <li className={styles.popoverLi}>
                          <BsReplyFill className={styles.popoverIcon} />
                          <p>Reply</p>
                        </li>
                        <li className={styles.popoverLi}>
                          <AiOutlineClose className={styles.popoverIcon} />
                          <p>Close</p>
                        </li>
                      </u>
                    }
                    trigger="click"
                  >
                    {<IoEllipsisHorizontal className={styles.horizButton} />}
                  </Popover>
                </td>
              </tr>
            ))}
          </tbody>
        )}
      </table>

      {!Boolean(data?.length) && (
        <div className="w-full h-full flex flex-col items-center justify-center p-8">
          <div>
            <Image src={EmptyDataTable} alt="empty table" />
          </div>
          {loading ? (
            <div>Loading Data ...</div>
          ) : (
            <div className="text-center mt-3 bolder">No available data</div>
          )}
        </div>
      )}
    </div>
  );
};
