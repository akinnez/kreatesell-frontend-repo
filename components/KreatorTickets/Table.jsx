import { EmptyDataTable } from "utils";
import styles from "./Table.module.scss";
import Image from "next/image";
import { IoEllipsisHorizontal } from "react-icons/io5";
import { Popover } from "antd";
import { useState } from "react";
import { AiOutlineClose, AiOutlineEye } from "react-icons/ai";
import { BsReplyFill } from "react-icons/bs";
import moment from "moment";

const SuccessDiv = ({ text }) => {
  return (
    <div
      style={{
        background: "#4bf71625",
        borderRadius: "5px",
        width: "50px",
        textAlign: "center",
      }}
    >
      <p style={{ color: "#176300" }}>{text}</p>
    </div>
  );
};

const ErrorDiv = ({ text }) => {
  return (
    <div
      style={{
        background: "#f7211622",
        borderRadius: "5px",
        width: "50px",
        textAlign: "center",
      }}
    >
      <p style={{ color: "#f5222d" }}>{text}</p>
    </div>
  );
};
const PendingDiv = ({ text }) => {
  return (
    <div
      style={{
        background: "#1455e042",
        borderRadius: "5px",
        width: "80px",
        textAlign: "center",
      }}
    >
      <p style={{ color: "#002d8f" }}>{text}</p>
    </div>
  );
};

const header = [
  {
    title: "Ticket ID",
    key: "ticket_id",
  },

  {
    title: "Subject",
    key: "subject",
  },
  {
    title: "Date",
    key: "date_created",
  },
  {
    title: "Department",
    key: "department",
  },

  {
    title: "Status",
    key: "status",
  },
  // {
  //   title: "Actions",
  //   key: "actions",
  // },
];
export const Table = ({ data }) => {
  const [openPopOver, setOpenPopOver] = useState(false);
  return (
    <div className={styles.tableContainer}>
      <table className={styles.table}>
        <thead>
          <tr>
            {header?.map((item, i) => (
              <th key={i}>{item?.title}</th>
            ))}
          </tr>
        </thead>

        {Boolean(data?.length) && (
          <tbody className="t-body">
            {data?.map((item, i) => (
              <tr key={item?.id || i}>
                <td>{item.id}</td>
                <td>{item.heading}</td>
                <td>{moment(item.date).format("YYYY-MM-DD HH:mm:ss")}</td>
                <td>{item.department}</td>

                <td>
                  {item.status?.toLowerCase() === "closed" ? (
                    <SucccessDiv text="Closed" /> ? (
                      item.status?.toLowerCase() === "answered"
                    ) : (
                      <SuccessDiv text="Answred" />
                    )
                  ) : (
                    <PendingDiv text={"Pending"} />
                  )}
                </td>
                {/* <td>
                  <Popover
                    placement="bottomRight"
                    title={null}
                    content={
                      <ul className={styles.popoverUl}>
                        <li className={styles.popoverLi}>
                          <AiOutlineEye className={styles.popoverIcon} />
                          <p>View Ticket</p>
                        </li>
                        <li className={styles.popoverLi}>
                          <BsReplyFill className={styles.popoverIcon} />
                          <p>Reply</p>
                        </li>
                        <li className={styles.popoverLi}>
                          <AiOutlineClose className={styles.popoverIcon} />
                          <p>Close</p>
                        </li>
                      </ul>
                    }
                    trigger="click"
                  >
                    {<IoEllipsisHorizontal className={styles.horizButton} />}
                  </Popover>
                </td> */}
              </tr>
            ))}
          </tbody>
        )}
      </table>

      {!Boolean(data?.length) && (
        <div className="w-full h-full flex flex-col items-center justify-center p-8">
          <div>
            <Image src={EmptyDataTable} />
          </div>
        </div>
      )}
    </div>
  );
};
