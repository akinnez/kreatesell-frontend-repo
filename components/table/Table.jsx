import { EmptyDataTable } from "utils";
import styles from "./Table.module.scss";
import Image from "next/image";

export const Table = ({ header, data, loading }) => {
	return (
		<div className={styles.tableContainer}>
			<table className={styles.table}>
				<thead>
					<tr>
						{header?.map((item, i) => (
							<th key={i}>{item?.title}</th>
						))}
					</tr>
				</thead>

				{Boolean(data?.length) && (
					<tbody className="t-body">
						{data?.map((data, i) => (
							<tr key={data?.id || i}>
								{header?.map((item, i) =>
									item?.component ? (
										<td key={i}>
											{item.component({ item: data[item?.key], data })}
										</td>
									) : (
										<td key={i}>{data[item?.key]}</td>
									)
								)}
							</tr>
						))}
					</tbody>
				)}
			</table>

			{!Boolean(data?.length) && (
				<div className="w-full h-full flex flex-col items-center justify-center p-8">
					<div>
						<Image src={EmptyDataTable} />
					</div>
					{loading ? (
						<div>Loading Data ...</div>
					) : (
						<div className="text-center mt-3 bolder">No available data</div>
					)}
				</div>
			)}
		</div>
	);
};
