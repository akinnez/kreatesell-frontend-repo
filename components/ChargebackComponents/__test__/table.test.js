import { Table } from "../Table";
import renderer from "react-test-renderer";

const mockProps = {
  header: [
    { title: "mockHeader1" },
    { title: "mockHeader2" },
    { title: "mockHeader3" },
  ],
  data: [
    {
      id: 0,
      name: "mockName",
      desc: "mockDesc1",
      department: "mockDept1",
      status: "Open",
      date: "2022-01-20 10:26:44",
    },
    {
      id: 0,
      name: "mockName2",
      desc: "mockDesc2",
      department: "mockDept2",
      status: "Closed",
      date: "2022-01-20 10:26:44",
    },
    {
      id: 0,
      name: "mockName3",
      desc: "mockDesc3",
      department: "mockDept3",
      status: "Open",
      date: "2022-01-20 10:26:44",
    },
  ],
};

describe("Table : ", () => {
  it("renders a snapshot of <Table /> with mockProps when loading is false", () => {
    const table = renderer
      .create(<Table loading={false} {...mockProps} />)
      .toJSON();
    expect(table).toMatchSnapshot();
  });

  it("renders a snapshot of <Table /> with mockProps when loading is true", () => {
    const table = renderer
      .create(<Table loading={true} {...mockProps} />)
      .toJSON();
    expect(table).toMatchSnapshot();
  });
});
