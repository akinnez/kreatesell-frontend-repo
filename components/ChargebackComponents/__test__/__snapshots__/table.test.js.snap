// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table :  renders a snapshot of <Table /> with mockProps when loading is false 1`] = `
<div
  className="tableContainer"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          mockHeader1
        </th>
        <th>
          mockHeader2
        </th>
        <th>
          mockHeader3
        </th>
      </tr>
    </thead>
    <tbody
      className="t-body"
    >
      <tr>
        <td>
          0
        </td>
        <td>
          mockName
        </td>
        <td>
          mockDesc1
        </td>
        <td>
          mockDept1
        </td>
        <td>
          2022-01-20 10:26:44
        </td>
        <td>
          <div
            style={
              Object {
                "background": "#f7211622",
                "borderRadius": "5px",
                "textAlign": "center",
                "width": "50px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#f5222d",
                }
              }
            >
              Open
            </p>
          </div>
        </td>
        <td>
          <svg
            className="horizButton"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          0
        </td>
        <td>
          mockName2
        </td>
        <td>
          mockDesc2
        </td>
        <td>
          mockDept2
        </td>
        <td>
          2022-01-20 10:26:44
        </td>
        <td>
          <div
            style={
              Object {
                "background": "#f7211622",
                "borderRadius": "5px",
                "textAlign": "center",
                "width": "50px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#f5222d",
                }
              }
            >
              Open
            </p>
          </div>
        </td>
        <td>
          <svg
            className="horizButton"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          0
        </td>
        <td>
          mockName3
        </td>
        <td>
          mockDesc3
        </td>
        <td>
          mockDept3
        </td>
        <td>
          2022-01-20 10:26:44
        </td>
        <td>
          <div
            style={
              Object {
                "background": "#f7211622",
                "borderRadius": "5px",
                "textAlign": "center",
                "width": "50px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#f5222d",
                }
              }
            >
              Open
            </p>
          </div>
        </td>
        <td>
          <svg
            className="horizButton"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table :  renders a snapshot of <Table /> with mockProps when loading is true 1`] = `
<div
  className="tableContainer"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          mockHeader1
        </th>
        <th>
          mockHeader2
        </th>
        <th>
          mockHeader3
        </th>
      </tr>
    </thead>
    <tbody
      className="t-body"
    >
      <tr>
        <td>
          0
        </td>
        <td>
          mockName
        </td>
        <td>
          mockDesc1
        </td>
        <td>
          mockDept1
        </td>
        <td>
          2022-01-20 10:26:44
        </td>
        <td>
          <div
            style={
              Object {
                "background": "#f7211622",
                "borderRadius": "5px",
                "textAlign": "center",
                "width": "50px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#f5222d",
                }
              }
            >
              Open
            </p>
          </div>
        </td>
        <td>
          <svg
            className="horizButton"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          0
        </td>
        <td>
          mockName2
        </td>
        <td>
          mockDesc2
        </td>
        <td>
          mockDept2
        </td>
        <td>
          2022-01-20 10:26:44
        </td>
        <td>
          <div
            style={
              Object {
                "background": "#f7211622",
                "borderRadius": "5px",
                "textAlign": "center",
                "width": "50px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#f5222d",
                }
              }
            >
              Open
            </p>
          </div>
        </td>
        <td>
          <svg
            className="horizButton"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td>
          0
        </td>
        <td>
          mockName3
        </td>
        <td>
          mockDesc3
        </td>
        <td>
          mockDept3
        </td>
        <td>
          2022-01-20 10:26:44
        </td>
        <td>
          <div
            style={
              Object {
                "background": "#f7211622",
                "borderRadius": "5px",
                "textAlign": "center",
                "width": "50px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#f5222d",
                }
              }
            >
              Open
            </p>
          </div>
        </td>
        <td>
          <svg
            className="horizButton"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
